flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/CheckpointBarrierAligner.java

```diff
@@ -214,8 +233,9 @@ protected void onBarrier(int channelIndex) throws IOException {
 	}
 
 	@Override
-	public boolean processCancellationBarrier(CancelCheckpointMarker cancelBarrier) throws Exception {
+	public void processCancellationBarrier(CancelCheckpointMarker cancelBarrier) throws Exception {
 		final long barrierId = cancelBarrier.getCheckpointId();
+		List<Integer> channelsToUnblock = new ArrayList<>();
```

Comment:
Considering my above concerns, all the resume notification can be done inside `releaseBlocksAndResetBarriers` instead